lib/puppet/parser/functions/template.rb

```diff
@@ -23,8 +23,12 @@
         wrapper.result
       rescue => detail
         info = detail.backtrace.first.split(':')
-        raise Puppet::ParseError,
-          "Failed to parse template #{file}:\n  Filepath: #{info[0]}\n  Line: #{info[1]}\n  Detail: #{detail}\n"
+        message = []
+        message << _("Failed to parse template %{file}:") % { file: file }
+        message << _("  Filepath: %{file_path}") % { file_path: info[0] }
+        message << _("  Line: %{line}") % { line: info[1] }
+        message << _("  Detail: %{detail}") % { detail: detail }
```

Comment:
Curious what the reason for splitting this into multiple translations is. I vaguely remember seeing some discussion somewhere, but I'm not seeing anything in the PR comments, commit, or ticket.

Reply:
There is a guidelines page in confluence that asks that externalized strings generally be one sentence each, to strike the right balance between a units that make sense (whole sentences) while still maybe being reusable (fewer translations to pay for).

Reply:
Ah! Thanks! The commit message should probably mention splitting up a translation to follow the guidelines, since it only says "add translations to ParseError raises" right now.

Reply:
For translation its best to have single sentences that you use to build up paragraphs. That way you only pay to translate a given sentence once and can reuse it. Like the "Detail:" line is used in other places in the code as well.